TritonSE/TSE-Oktavian
# Added page for editing role permissions

### Administrative Info
Monday Board ID: 1181559666
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
Added a page to edit roles and create roles
Connected pages to corresponding backend route

### Testing
How did you confirm your changes worked? 
1) Went to the page to display all roles
2) clicked on the edit button for a role and checked that permissions are correctly displayed
3) edit permissions and clicked save
4) Went back to page that displays all roles and clicked the "+ create role" button
5) Create role and gave random permissions (also tested repeat role name and empty role name)
6) Deleted role I just created by editing the role and clicking the delete button

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

Default permissions for developers
![image](https://user-images.githubusercontent.com/55808666/118589486-53be0f80-b755-11eb-9340-02bdb058d752.png)


Permissions after saving changes
![image](https://user-images.githubusercontent.com/55808666/118589462-47d24d80-b755-11eb-97d3-befcb781870a.png)


Creating a role named Test (code will auto-capitalize role name)
![image](https://user-images.githubusercontent.com/55808666/118589570-7a7c4600-b755-11eb-9880-a47d17def124.png)


Role appears after being created
![image](https://user-images.githubusercontent.com/55808666/118589597-89fb8f00-b755-11eb-8bb4-38c43928e690.png)


Deleting the Test role
![image](https://user-images.githubusercontent.com/55808666/118589645-a1d31300-b755-11eb-804b-aee77103f60a.png)


Test role gone after being deleted
![image](https://user-images.githubusercontent.com/55808666/118589688-b6afa680-b755-11eb-8d31-c312f235633a.png)
